app/pages/notifications/notification-section.jsx

```diff
@@ -117,6 +118,15 @@ export default class NotificationSection extends Component {
     });
   }
 
+  markAllRead() {
+    talkClient.put('/notifications/read', { section: this.props.section });
+    this.state.notificationData.forEach((data) => {
+      data.notification.update({ delivered: true }).save();
+    });
+    this.setState({ unread: 0 });
+    this.context.notificationsCounter.setUnread(0);
```

Comment:
You can't really use this to clear the counter since you're only marking one section as read -- e.g. you don't know if the other sections contain unread or not.